docs/guides/gateway/gateway_general_documentation.md

```diff
@@ -0,0 +1,28 @@
+
+# Gateway General Documentation
+
+The Shifu project's gateway is a multi-protocol bridge that supports various protocols such as HTTP, MQTT and LwM2M. It provides a unified interaction interface for different IoT devices and systems, simplifying protocol conversion and integration.
+
+## Overview
+
+The LwM2M (Lightweight Machine to Machine) Gateway consists of two main components: an LwM2M client that connects to the LwM2M server and an HTTP client that connects to deviceShifu.   The LwM2M Gateway enables deviceShifu to adapt to the LwM2M protocol, handle requests from the LwM2M server, and push data to the cloud, fulfilling the two major functions required by the LwM2M protocol: the capability to pull data from the device and post data from the cloud to the device.
```

Comment:
Gateway is a concept, and LwM2M Gateway is one of the implementations based on this concept. it likes deviceShifu and deviceshifu-HTTP.

Reply:
Need to make sense it before write this doc